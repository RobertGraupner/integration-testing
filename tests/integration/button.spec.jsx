// obiekt screen służy do przechowywania referencji do elementów DOM
import { render, screen, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom"
import { Button } from '../../src/components/Button/Button';

test('Properly renders passed text', () => {
  // render zawiera container, który zawiera wszystkie elementy wyrenderowane w teście
  const { container } = render(<Button>Click me</Button>);
  // sprawdzamy czy przycisk zawiera tekst 'Click me'
  expect(screen.getByRole('button')).toHaveTextContent('Click me');
  // snapshot jest to zrzut zawartości containera, który jest porównywany z poprzednim zrzutem
  // Jeśli coś zmienimy w komponencie to snapshot się nie zgadza i test się nie powiedzie
  expect(container).toMatchSnapshot();
});

test('Calls onClick callback', () => {
  // tworzymy funkcję, która będzie nasłuchiwać na kliknięcie
  const onClick = jest.fn(() => console.log('Button clicked'));
  render(<Button onClick={onClick}>Click me</Button>);
  // klikamy na przycisk
  fireEvent.click(screen.getByRole('button'));
  // sprawdzamy czy funkcja onClick została wywołana
  expect(onClick).toHaveBeenCalled();
});
