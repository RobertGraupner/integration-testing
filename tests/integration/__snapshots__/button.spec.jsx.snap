// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properly renders passed text 1`] = `
<div>
  <button
    class="button"
  >
    Click me
  </button>
</div>
`;
